src/net.cpp

```diff
@@ -1688,8 +1722,17 @@ void ThreadOpenConnections()
             break;
         }
 
-        if (addrConnect.IsValid())
-            OpenNetworkConnection(addrConnect, (int)setConnected.size() >= std::min(nMaxConnections - 1, 2), &grant);
+        if (addrConnect.IsValid()) {
+
+            if (fFeeler) {
+                // Add small amount of random noise before connection to avoid syncronization.
+                int randsleep = GetRandInt(FEELER_SLEEP_WINDOW * 1000);
```

Comment:
Would it make sense to use PoissonNextSend here, resulting in maximally unpredictable times?